torch/csrc/autograd/python_legacy_variable.cpp

```diff
@@ -48,21 +48,22 @@ static PyObject *THPVariable_pynew(PyTypeObject* type, PyObject *args, PyObject
     // by nn.Parameter() with no arguments.
     auto scalar_type = torch::tensors::get_default_scalar_type();
     auto var = at::empty({0}, torch::tensors::get_default_tensor_type().options(scalar_type));
-    tensor = static_cast<Variable&>(var).tensor_data();
+    tensor = static_cast<Variable&>(var).variable_data();
   } else if (THPVariable_Check(data)) {
-    tensor = ((THPVariable*)data)->cdata.tensor_data();
+    tensor = ((THPVariable*)data)->cdata.variable_data();
   } else {
     throw torch::TypeError("Variable data has to be a tensor, but got %s",
         Py_TYPE(data)->tp_name);
   }
+  tensor.unsafeGetTensorImpl()->set_allow_tensor_metadata_change(true);
```

Comment:
We set `tensor`'s `allow_tensor_metadata_change` to true here, because we want to allow the following use case:
```python
var = Variable(torch.randn(2, 3))
var.resize_(4, 5)
```
for backward compatibility, which is used by tests such as `TestAutograd.test_resize_as_`.

Reply:
can you please, consistently, put comments like this in the code rather than the review?

Also, what is the difference between this and `detach`?